lib/provisioning/ProvisionDispatcher.js

```diff
@@ -28,32 +27,37 @@ class ProvisionDispatcher {
      * @constructor
      * @param {object} zkConfig - zookeeper config object
      * @param {string} zkConfig.connectionString - zookeeper connection string
+     * @param {string} [_zookeeper] - zookeeper module
+     * @param {boolean} [_runningCI] - introduce random delays
      * where provisions are managed
      * (e.g. "localhost:2181/shared-tasks")
      */
-    constructor(zkConfig) {
+    constructor(zkConfig, _zookeeper, _runningCI = false) {
```

Comment:
I think splitting functionality by running in a unit test or not kinda strange. Maybe rename this to `maxRandomDelay` or introduce a "private" method to add this function outside the constructor.

Reply:
I could not remove `_zookeeper` since it is used in the constructor.